### PR TITLE
Jwilber/update esm2 native te.yaml

### PR DESCRIPTION
**Pull request by jwilber**
Update `esm2_native_te` config to have better comments and structure, with better support for overriding script keys and observability stuff to track.

Also propagate the `cfg.num_nodes` to parallelism key in `LeptonJobUserSpec` (this was unclear in the python sdk instructions).

## Summary by CodeRabbit

- New Features
  - Expanded recipe surface with explicit identifiers (variant, framework, precision, parallelism strategy), TE/FP8 toggles, wandb_init_args, script_args, products list, and run_script orchestration.
- Improvements
  - Script-level controls for train steps, micro-batch size, warmup and checkpointing; run commands receive tracking and checkpoint parameters.
  - Job parallelism now follows configured node/device counts; job names prefixed for clearer tracing; enhanced experiment tracking (mode/project/group/job_type/name).
- Removed
  - Removed a legacy top-level config key from the base configuration.
- Chores
  - Removed non-essential debug shell output.

### CODE REVIEW

File: ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml [hunk 1]
```diff
@@ -3,20 +3,39 @@ defaults:
...
 node_group: yo-bom-lepton-001
 num_nodes: 2
 device_type: gpu
-num_devices: 2
+num_devices: 8
 gpu_type: h100-sxm
-total_gpus: ${multiply:${num_devices},${num_nodes}}
 resource_shape: "${device_type}.${num_devices}x${gpu_type}"
+# `total_gpus` calculated from above
+total_gpus: ${multiply:${num_devices},${num_nodes}}
```
Comment (dorotat-nv) [inline ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml]: where total_gpus is used?

File: ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml [hunk 2]
```diff
@@ -28,21 +47,42 @@ trainer:
...
     wandb_name: "${config}__${now:%Y%m%d-%H%M%S}__${gitsha:}"
 
-# training script to run
+############################################################
+# run script
+# This gets called right after `checkout_script` in the base config.
+############################################################
 run_script: |
-  accelerate launch --config_file accelerate_config/default.yaml \
+  accelerate launch --config_file accelerate_config/${acc_config}.yaml \
```
Comment (dorotat-nv) [inline ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml:81]: would  the train command auto discover number of available gpus ? we don't need to pass it to the training script?

Comment (dorotat-nv) [reply ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml:81]: I would reccomend to control mbs, number of gpus and precisions as well as features from the config ie to pass from this config even if the settings from acc_config are in agreements. It will ensure things will not diverge

File: ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml [hunk 3]
```diff
@@ -3,46 +3,100 @@ defaults:
...
     --config-name ${config}.yaml \
-    num_train_steps=${num_train_steps} \
-    wandb_init_args.mode=${wandb_init_args.mode} \
-    +wandb_init_args.project=${wandb_init_args.project} \
+    +wandb_init_args.mode=${wandb_init_args.mode} \
+    wandb_init_args.project=${wandb_init_args.project} \
     +wandb_init_args.group=${wandb_init_args.group} \
     +wandb_init_args.job_type=${wandb_init_args.job_type} \
-    wandb_init_args.name=${wandb_name}
+    wandb_init_args.name=${wandb_name} \
```
Comment (dorotat-nv) [inline ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml:94]: yes, exacly this I had in mind in my previous comment! can we pass here a setting for the precision maybe?

File: ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml [hunk 4]
```diff
@@ -3,46 +3,100 @@ defaults:
...
-# wandb
+# Catchall for additional features/configs
+extras: [] # e.g. [thd]
+
+############################################################
+# wandb info (total_gpus used for group name)
+############################################################
 wandb_init_args:
   project: "test_convergence__recipes__${sanitize:${branch}}"
   group: "${model_type}__${task_cmd}__${total_gpus}__${sanitize:${gpu_type}}"
```
Comment (dorotat-nv) [inline ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml]: maybe  `_${total_gpus}gpus_` since it might be hard to decode from the group name itself

File: ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml (hunk 1 above)
Comment (jwilber) [reply ci/lepton/model_convergence/configs/recipes/esm2_accelerate_te.yaml]: wandb info

File: ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml (hunk 4 above)
Comment (jwilber) [reply ci/lepton/model_convergence/configs/recipes/esm2_native_te.yaml]: that's a good idea

**Review by trvachov — APPROVED** [review]
